s3-connector/unittest_s3-connector.py

```diff
@@ -20,62 +20,96 @@ def setUpClass(self):
         self.secret= self.test_cfg['secret']
         self.s3_endpoint=self.test_cfg['s3_endpoint']
         self.region_name=self.test_cfg['region_name']
-        # Intialize connection for S3
+       
         self.s3_connection = s3connector.S3(
             self.test_cfg["key"], 
             self.test_cfg["secret"], 
             self.test_cfg["s3_endpoint"], 
             self.test_cfg["region_name"]
             )
 
-    # Test connection for S3
+    # test connection for s3 resource
     def test_connection(self):
         self.assertTrue(self.s3_connection)
+        
+    #  test connection with incorrect keys
+    def test_conn_with_incorrectkey(self):
+        try:
+          fun = s3connector.S3( 
+            self.test_cfg["incorrect_key"],
+            self.test_cfg["incorrect_secret"],
+            self.test_cfg["s3_endpoint"],
+            self.test_cfg["region_name"]
+            ).list_buckets()
+          for bucket in fun:
+            print(bucket)
+        except ClientError as e:
```

Comment:
To test exception use:
```python:
        with self.assertRaises(ClientError):
            fun = s3connector.S3( 
            self.test_cfg["incorrect_key"],
            self.test_cfg["incorrect_secret"],
            self.test_cfg["s3_endpoint"],
            self.test_cfg["region_name"]
            ).list_buckets()

```
Example: see test_unsupported_connector test case in https://github.com/cnvrg/unittest-framework/blob/main/connectorUnittest/test_connector_factory.py

Reply:
@orionsBeltWest  As recommened, updated the test exception.